src/Filter/CallbackFilter.php

```diff
@@ -29,7 +29,18 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $value
             throw new \RuntimeException(sprintf('Please provide a valid callback option "filter" for field "%s"', $this->getName()));
         }
 
-        $this->active = \call_user_func($this->getOption('callback'), $queryBuilder, $alias, $field, $value);
+        $isActive = \call_user_func($this->getOption('callback'), $queryBuilder, $alias, $field, $value);
+        if (!\is_bool($isActive)) {
+            @trigger_error(
+                'Using another return type than boolean for the callback option is deprecated'
+                .' since sonata-project/doctrine-orm-admin-bundle 3.x and will throw an exception in version 4.0.',
+                E_USER_DEPRECATED
+            );
+
+            // NEXT_MAJOR: Throw an Exception instead.
```

Comment:
I think it's easier to define what exception would be now that we have context and just uncomment it in `4.x`. 

This would ease removing it in `4.x`, maybe someone else does it and has to read the whole thing to figure out.

Reply:
It should be ok now